tests/index.test.js

```diff
@@ -0,0 +1,144 @@
+  return fastify
+}
+
+const MONGODB_URL = `mongodb://${(process.env.MONGO_HOST || 'localhost')}:27017/test`
+t.test('%CUSTOM_PLUGIN_SERVICE_NAME%', async t => {
+  // silent => trace for enabliing logs
+  const fastify = await setupFastify({
+    USERID_HEADER_KEY: 'userid',
+    GROUPS_HEADER_KEY: 'groups',
+    CLIENTTYPE_HEADER_KEY: 'clienttype',
+    BACKOFFICE_HEADER_KEY: 'backoffice',
+    MICROSERVICE_GATEWAY_SERVICE_NAME: 'microservice-gateway.example.org',
+    MONGODB_URL,
+  })
+
+  t.tearDown(async() => {
+    await fastify.close()
+  })
+
+  t.test('POST /greetings', t => {
+    t.test('ok', async t => {
+      t.plan(2)
+      await removeAllElementFromMyCollection()
+      const response = await fastify.inject({
+        method: 'POST',
+        url: '/greetings',
+        headers: {
+          userid: 'my-user-id',
+        },
+        payload: {
+          who: 'Foo',
+        },
+      })
+      t.equal(response.statusCode, 204)
+
+      t.test('should insert a document', async t => {
+        t.plan(4)
+        const collection = fastify.mongo.db.collection('mycollection')
+        const allDocs = await collection.find({}).toArray()
+
+        t.equal(allDocs.length, 1)
+        t.strictSame(allDocs[0].from, 'my-user-id')
+        t.strictSame(allDocs[0].to, 'Foo')
+        t.strictSame(allDocs[0].type, 'hello')
+      })
+    })
+    t.end()
+  })
+
+  t.test('GET /greetings', async t => {
+    await removeAllElementFromMyCollection()
+
+    t.test('no greetings found', async t => {
+      t.plan(2)
+
+      const response = await fastify.inject({
+        method: 'GET',
+        url: '/greetings',
+        headers: {
+          userid: 'no-existing-user',
+        },
+      })
+      t.equal(response.statusCode, 404)
+      t.strictSame(response.payload, 'No greetings found')
+    })
+
+    t.test('greetings found', async t => {
+      t.plan(2)
+
+      await fastify.mongo.db.collection('mycollection').insertOne({ from: 'my-user-id', type: 'hello', to: 'Foo' })
+
+      const response = await fastify.inject({
+        method: 'GET',
+        url: '/greetings',
+        headers: {
+          userid: 'my-user-id',
+        },
+      })
+      t.equal(response.statusCode, 200)
+      t.same(JSON.parse(response.payload), {
+        from: 'my-user-id',
+        to: 'Foo',
+        type: 'hello',
+      })
+    })
+
+    t.test('error connecting to Mongo using not existing host', async t => {
+      t.plan(1)
+      const WRONG_MONGODB_URL = `mongodb://wronghost:27017/test`
+      t.rejects(setupFastify({
+        USERID_HEADER_KEY: 'userid',
+        GROUPS_HEADER_KEY: 'groups',
+        CLIENTTYPE_HEADER_KEY: 'clienttype',
+        BACKOFFICE_HEADER_KEY: 'backoffice',
+        MICROSERVICE_GATEWAY_SERVICE_NAME: 'microservice-gateway.example.org',
+        MONGODB_URL: WRONG_MONGODB_URL,
+      }))
+    })
+    t.test('error contacting Mongo after close connection', async t => {
+      t.plan(1)
+      await fastify.close()
```

Comment:
This command close fastify server, so it is not possible to contact mongo. Could be interesting to see when mongo connection close.